src/MergeContext.js

```diff
@@ -954,12 +968,47 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // recalculates this._stagingFailed value for PRs without staging commit
+    async _recalculateStagingFailed() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // we consider all commits created by the bot user and referencing this PR as staged commits
+        let stagedEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
```

Comment:
I assume there are many PR events that are not related to commit creation. Let's assume that future Anubis versions will create such events as well, even if they do not do that today. Should not our filter expression discard events that are not related to a commit creation? Or does `ev.event === "referenced"` already do that reliably? In other words, does `ev.event === "referenced"` imply that `ev` is a commit creation event and not some other event that "referenced" something?

```suggestion
        let stagingEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
```

Reply:
>  Or does ev.event === "referenced" already do that reliably?

According to the [documentation](https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#referenced), the "referenced' attribute means "The issue was referenced from a commit message.". This should be quite reliable - as long as only Anubis creates commits referencing PRs.